packages/typespec-azure-resource-manager/lib/models.tsp

```diff
@@ -7,6 +7,30 @@ using Azure.ResourceManager.Private;
 
 namespace Azure.ResourceManager;
 
+model DefaultNameParameter<
```

Comment:
Per discussion, if we rejigger the decorator slightly, so that they take a keyname and segment name, and the name being set to null or never means that they calculate the default (and otherwise, they use the supplied name).  Should allow us to consolidate these templates.

Reply:
Note that this also makes it easier to change one thing (say the pattern) without having to change others